rosette/src/RBLstream.cc

```diff
@@ -348,7 +348,7 @@ DEF("ostream-log-time", obLogTime, 1, 1) {
         timeinfo = localtime(&rawtime);
         strftime(buf, sizeof(buf), "%Y-%m-%d %I:%M:%S\n", timeinfo);
 
-        if (errno = fprintf(strm->stream, "%s", buf)) {
+        if ((errno = fprintf(strm->stream, "%s", buf))) {   // Assign and test
```

Comment:
No assignments in conditionals.

Reply:
Looking at this more closely, I don't understand why he is setting the global errno variable with the result of the fprintf.  fprintf returns the length of the output, or a negative number if there's an error. The man page makes no reference to errno. Regardless, I don't think he's handling the return value correctly. He's testing for non-zero, which may or may not be negative. If non-zero, he returns FIXNUM(-errno).

Any suggestions what I should return here?  My guess is PRIM_ERROR on <= 0, and NIV on success. Anything I do is going to be a change in operation.

Reply:
As of C++11, errno is a `thread-local`, not a `static` global. But, regardless, I think this pattern is just "wrong." If you replace errno with a stack variable, I can't see how anything changes here. But, the code does clearly return something like `-10` if it successfully wrote 10 characters to the output stream. I don't know what the Rosette specification is, but this does seem a bit strange.

What is the `NIV` value's meaning?

Reply:
`NIV` means no-intrinsic-value and is used when no other value is appropriate.

Reply:
I think we resolved to not use `errno` because that's just weird, but you probably want to keep the strange behavior of returning the negative of whatever fprintf returns, since in `Vm.cc` it looks like return values end up in VM registers.

Reply:
**Done**